### PR TITLE
fixed accessing model.val_dataloader to model.trainer.val_dataloader

### PR DESCRIPTION
**Pull request by danielgafni**
### Description

fixing #757 

This PR ...

### Checklist

- [x] Linked issues (if existing)
- [ ] Amended changelog for large changes (and added myself there as contributor)
- [ ] Added/modified tests
- [x] Used pre-commit hooks when committing to ensure that code is compliant with hooks. Install hooks with `pre-commit install`.
      To run hooks independent of commit, execute `pre-commit run --all-files`

Make sure to have fun coding!

### CODE REVIEW

**Comment by danielgafni** [comment]
Hmm, looks like accessing dataloaders from `configure_optimizers` can't be done until this pytorch-lightning [issue](https://github.com/PyTorchLightning/pytorch-lightning/issues/10430) is resolved

**Comment by jdb78** [comment]
Looks like the team is actively working on it.

**Comment by jdb78** [comment]
We should be able to fix it by removing the trainer dependency and making the lr scheduler optional, i.e. `strict=False`
